fix(ci): check, create and mount lwd cached state for tests

arya2 opened the pull request:
## Motivation

This PR initially aimed to add checks for the existence of the lightwalletd cached state directory (`LWD_CACHE_DIR`) before running integration tests that require it. During development and testing, several related issues in the CI workflow and test harness were identified and fixed:

1.  Tests requiring a pre-existing `lightwalletd` state (`lwd-update-sync`, `lwd-send-transactions`, `lwd-grpc-wallet`) were timing out. This was traced back to the `lwd-full-sync` test not correctly saving its state to the expected `LWD_CACHE_DIR`, preventing the CI from creating valid `lwd-cache-*` disk images.
2.  The test harness code responsible for launching `lightwalletd` contained debug logging that caused a panic during the `lwd-full-sync` test by attempting to read the state directory before `lightwalletd` created it.
3.  Inconsistencies in constructing disk image names/labels were observed, related to recent updates in the `rlespinasse/github-slug-action` in their v5.

Fixes #9419

## Solution

1.  **CI Workflow (`sub-ci-integration-tests-gcp.yml`, `sub-deploy-integration-tests-gcp.yml`, `sub-find-cached-disks.yml`):**
    *   Added the missing `-e LWD_CACHE_DIR=/home/zebra/.cache/lwd` environment variable to the `lwd-full-sync` test definition in `sub-ci-integration-tests-gcp.yml`. This ensures `lightwalletd` saves its state to the correct location.
    *   The `create-state-image` step in `sub-deploy-integration-tests-gcp.yml` will now correctly capture this state when creating `lwd-cache-*` images.
    *   Updated workflow files to use `env.GITHUB_REF_POINT_SLUG_URL` instead of `env.GITHUB_REF_SLUG_URL` for disk/label naming consistency, to mitigate the issues with `github-slug-action@v5`. (See [rlespinasse/github-slug-action#104](https://github.com/rlespinasse/github-slug-action/issues/104)).
    *   Added `github-slug-action` to `sub-find-cached-disks.yml` to ensure consistent variable availability.

2.  **Test Harness (`zebrad/tests/common/lightwalletd.rs`):**
    *   Modified `spawn_lightwalletd_child` to accept the `TestType` as an argument.
    *   Conditionalized the debug logging that reads the `lightwalletd` state directory size. It now skips this check for `TestType::FullSyncFromGenesis` (used by `lwd-full-sync`) to prevent panics when the directory doesn't initially exist.

3.  **RPC (`zebra-rpc/src/methods.rs`, `zebra-rpc/src/methods/get_block_template_rpcs.rs`):**
    *   Modified `chain_tip_difficulty` to accept a `should_use_default` flag. When true (as used by `getinfo`), it returns a default difficulty value instead of erroring if the state service isn't ready, preventing a potential panic.

### Tests

The primary testing for these changes involves the CI integration tests defined in `sub-ci-integration-tests-gcp.yml`:

-   The fix to `lwd-full-sync` saving its state correctly is verified by the subsequent successful runs of `lwd-update-sync`, `lwd-send-transactions`, and `lwd-grpc-wallet`, which depend on the `lwd-cache-*` disk image created by `lwd-full-sync`. These tests should no longer time out due to missing state.
-   The fix in `lightwalletd.rs` is verified by the `lwd-full-sync` test no longer panicking when attempting to log the cache directory size.

### Specifications & References

-   GitHub Slug Action Issue: [GITHUB_REF_NAME is not correctly filled on pull_request events #104](https://github.com/rlespinasse/github-slug-action/issues/104)

### PR Checklist

<!-- Check as many boxes as possible. -->

- [x] The PR name is suitable for the release notes.
- [x] The solution is tested (via CI integration tests).
- [x] The documentation is up to date (Internal test harness/CI documentation updated implicitly via code comments).

arya2 committed:
fixes a bug in the `getinfo` RPC

arya2 committed:
logs the `--data-dir` arg passed to lwd in tests and the size of the directory.

arya2 committed:
compilation fixes

gustavovalverde committed:
fix(ci): save the lwd state in a pre-defined location

gustavovalverde committed:
fix(tests): avoid panic when logging for `lightwalletd_full_sync`

The debug logging code added to `spawn_lightwalletd_child` assumed that the `db/main` subdirectory would always exist within the provided `lightwalletd_state_path`. This caused a panic during the `lightwalletd_full_sync` test (which uses `TestType::FullSyncFromGenesis`), as this test provides the path where `lightwalletd` *will create* the state, but the directory doesn't exist at the time the logging code runs.

This commit wraps the directory reading and size logging logic in an `if !matches!(test_type, TestType::FullSyncFromGenesis { .. })` check. This ensures the logging only occurs for test types where the directory is expected to already exist, preventing the panic during full sync tests while retaining the logging for update tests.

gustavovalverde committed:
chore: fix build issue

gustavovalverde committed:
fix(workflows): always use the same name for disks in PR/branches

gustavovalverde commented:
I confirmed that:

Lightwalletd tests fail when the lwd state is not available: 
- https://github.com/ZcashFoundation/zebra/actions/runs/14820065858/attempts/2?pr=9481

`lwd-full-sync` can be run, and it does not trigger the failure:
- https://github.com/ZcashFoundation/zebra/actions/runs/14810607881/job/41584899715

All lwd tests pass when the state is available:
- https://github.com/ZcashFoundation/zebra/actions/runs/14820065858/job/41611470826?pr=9481

upbqdn commented on zebrad/tests/common/lightwalletd.rs:
```diff
@@ -207,13 +211,39 @@ where
+
+                tracing::info!("{lwd_cache_dir_size} bytes in lightwalletd cache dir");
+
+                let lwd_cache_dir =
```
We get the same dir above at line 219, could we not clone that? Or should we remove the whole `matches!` section now?

gustavovalverde replied:
Solved here:
- [refactor(tests): Avoid redundant directory read for lightwalletd cach…](https://github.com/ZcashFoundation/zebra/pull/9481/commits/f4b328253391c30dd74773750bd25872c4ca3754)

upbqdn replied:
Great, thanks. Do you or @arya2 want to keep this code? It looks like it's suited only for debugging.

gustavovalverde replied:
It's helpful. We previously had this in the entrypoint and it help catching (and understanding) this type of issues

gustavovalverde committed:
refactor(tests): Avoid redundant directory read for lightwalletd cache logs

Previously, `std::fs::read_dir` was called twice within the `spawn_lightwalletd_child` function when logging the size and contents of the lightwalletd cache directory if it already existed.

This change collects the directory entries into a `Vec` first, then iterates over the vector to calculate the size and log the individual entries, avoiding the second filesystem access.

gustavovalverde committed:
fix: compile issue

upbqdn reviewed (commented):
Was this the ultimate bug?

Edit: I meant it for commit [373e0a9](https://github.com/ZcashFoundation/zebra/pull/9481/commits/373e0a92f89f2527db90a6d78b441e5eea33bc9c).

gustavovalverde commented:
> Was this the ultimate bug?
> 
This missing variable https://github.com/ZcashFoundation/zebra/pull/9481/files#diff-88f0a9cbc54ea70f51d3d7163ed664a292b5d5e5b3bbcf79d6bda582e46f9638R319

The cached state was being recreated in a different location.

gustavovalverde reviewed (approved):
I'm re-approving to get this merged.

gustavovalverde commented:
@upbqdn it seems it will need your approval as I pushed commits to the branch :)

conradoplg commented:
@mergifyio queue

str4d commented on zebra-rpc/src/methods.rs:
```diff
@@ -2702,7 +2704,11 @@ mod opthex {
+pub async fn chain_tip_difficulty<State>(
+    network: Network,
+    mut state: State,
+    should_use_default: bool,
```
This new field is undocumented, and I do not know how to set it. (This method is used downstream in Zaino.)
